pkg/repositories/joke.go

```diff
@@ -0,0 +1,124 @@
+package repositories
+
+import (
+	"chuck-jokes/models"
+	gormModels "chuck-jokes/pkg/database/gorm/models"
+	"log"
+	"time"
+
+	"gorm.io/gorm"
+)
+
+type JokeRepository interface {
+	JokeOfTheDay(time string) *models.Joke
+	GetJoke(jokeID uint) *models.Joke
+	GetStatistic(jokeID uint) (*models.Joke, uint)
+	GetJokes(page int, perPage int) *Pagination[models.Joke]
+	JokeExistInLastMonth(joke *models.Joke) bool
+	GetFavourites(page int, perPage int, userID uint) *Pagination[models.Joke]
+}
+
+// Joke base joke repository
+type Joke struct {
+	db *gorm.DB
+}
+
+// NewJoke create new joke repository
+func NewJoke(db *gorm.DB) *Joke {
+	return &Joke{
+		db: db,
+	}
+}
+
+// JokeOfTheDay get joke of the day from specify day
+func (j *Joke) JokeOfTheDay(time string) *models.Joke {
+	var joke = models.Joke{}
+	j.db.Where("created_at >= ?", time).First(&joke)
+
+	if joke.ExternalID == "" {
+		return nil
+	}
+	return &joke
+}
+
+// GetJoke get joke of the day from specyfi day
+func (j *Joke) GetJoke(jokeID uint) *models.Joke {
+	var joke = models.Joke{}
+	txFind := j.db.First(&joke, jokeID)
+
+	if txFind.Error != nil {
```

Comment:
ten error mozna zmielic jak pokazalem wyzej gdzies w przykladzie - mozesz schowac to j.db.First do ifa